### PR TITLE
update to go1.12

### PR DESCRIPTION
**Pull request by kyessenov**
/assign @utka 
(will need to follow up with prow update)
Signed-off-by: Kuat Yessenov <kuat@google.com>

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ version: 2
 defaults: &defaults
   working_directory: /go/src/istio.io/istio
   docker:
-    - image: istio/ci:go1.11-k8s1.10.4-helm2.7.2-minikube0.25
+    - image: gcr.io/istio-testing/circleci:2019-04-22
```
Comment (geeknoid) [inline .circleci/config.yml]: Where is this image built from?

Comment (kyessenov) [reply .circleci/config.yml]: .circleci/Dockerfile

**Comment by kyessenov** [comment]
IMHO we need a canonical build image that is shared by CI and developers. However, I'm not sure how to unwind all the dependencies within circleci Dockerfile. It seems to use an obsolte k8s version, in particular.

**Comment by kyessenov** [comment]
Seems like go1.12 changed how tests are ran which breaks code coverage task.

**Comment by kyessenov** [comment]
Folks we need to force merge over coverage. The new test runner in go 1.12 will treat logging after test ends as panic, which breaks coverage job, since it runs a previous version of the code.

**Comment by kyessenov** [comment]
1. We don't have to use docker hub, right? We can create a new Google project that has tighter access control to prevent write access.
2. Please share docker hub credentials. The only reason I used gcr.io/istio-testing is because I don't have access to docker hub.

**Comment by fpesce** [comment]
Hey @geeknoid any preferred approach here? New account on gcr.io ? istio-release for example.

@costinm has a very good point in #14306 , I'm not comfortable in a release build process that pulls images from non-trusted environment. This includes nightly/daily/CI builds that could be considered as prod candidates by Istio users.

**Pull request merged**